File: q3484/q3484.py
class Spreadsheet:

    def __init__(self, rows: int):
        self.grid = [[0 for _ in range(26)] for _ in range(rows)]
        
    def setCell(self, cell: str, value: int) -> None:
        (col, row) = self.getPos(cell)
        self.grid[row][col] = value     

    def resetCell(self, cell: str) -> None:
        (col, row) = self.getPos(cell)
        self.grid[row][col] = 0 

    def getValue(self, formula: str) -> int:
        parts = formula[1:].split("+")
        return sum(self.evaluateParts(part.strip()) for part in parts)
    
    def getPos(self, cell: str) -> Tuple[int, int]:
        col = ord(cell[0]) - ord('A')
        row = int(cell[1:]) - 1
        return (col, row)

    def evaluateParts(self, expression: str) -> int:
        if expression.isdigit():
            return int(expression)
        else:
            (col, row) = self.getPos(expression)
            return self.grid[row][col]

File: q3484/q3484.java
class Spreadsheet {
    private Map<String, Integer> map;

    public Spreadsheet(int rows) {
        map = new HashMap<>();
    }
    
    public void setCell(String cell, int value) {
        map.put(cell, value);
    }
    
    public void resetCell(String cell) {
        map.put(cell, 0);
    }
    
    public int getValue(String formula) {
        formula = formula.substring(1);
        String[] parts = formula.split("\\+");
        int left =  Character.isUpperCase(parts[0].charAt(0)) ? 
                    map.getOrDefault(parts[0], 0) :
                    Integer.parseInt(parts[0]); 
        int right = Character.isUpperCase(parts[1].charAt(0)) ? 
                    map.getOrDefault(parts[1], 0) :
                    Integer.parseInt(parts[1]);
        return left + right;
    }
}

/**
 * Your Spreadsheet object will be instantiated and called as such:
 * Spreadsheet obj = new Spreadsheet(rows);
 * obj.setCell(cell,value);
 * obj.resetCell(cell);
 * int param_3 = obj.getValue(formula);
 */